src/config.js

```diff
@@ -36,7 +36,7 @@ export const urls = {
   thirdPartyFrameRegex: thirdPartyFrameRegex || /^d-\d+\.ampproject\.net$/,
   cdn: env['cdnUrl'] || 'https://cdn.ampproject.org',
   cdnProxyRegex: cdnProxyRegex ||
-      /^https:\/\/([a-zA-Z0-9_-]+\.)?cdn\.ampproject\.org/,
+      /^https:\/\/([a-zA-Z0-9_-]+\.)?cdn\.ampproject\.org$/,
```

Comment:
This will make the regex no longer match foo.cdn.ampproject.org.badguys.com, and also make it stop matching foo.cdn.ampproject.org/some/legit/file.html
Searched the code for use of this regex, and aside from a stub, all I found was the following, which is testing an origin (e.g. no path/filename) against the regex: https://github.com/ampproject/amphtml/blob/d230337dd356276492d9255c2cc2ce6e4dbce97d/src/url.js#L315
So, at least at this point in time, this shouldn't hurt anything. If in the future we want to test full URLs to see if they come from the CDN, then we should change this to allow an optional slash and path.